src/test/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientTest.java

```diff
@@ -414,6 +415,58 @@ public void testDropChannel() throws Exception {
             ArgumentMatchers.refEq("drop channel"));
   }
 
+  @Test
+  public void testDropChannelRequestNoLoggingLine() throws Exception {
+    DropChannelResponse responseSuccess = new DropChannelResponse();
+    responseSuccess.setStatusCode(RESPONSE_SUCCESS);
+    responseSuccess.setMessage("dropped");
+    String responseStringSuccess = objectMapper.writeValueAsString(responseSuccess);
+
+    ChannelsStatusResponse responseFailure = new ChannelsStatusResponse();
+    responseFailure.setStatusCode(0L);
+    responseFailure.setMessage("honk");
+    responseFailure.setChannels(new ArrayList<>());
+    String responseStringFailure = objectMapper.writeValueAsString(responseFailure);
+
+    String expectEntityFormat =
+        "{\"request_id\":\"%s\",\"role\":null,\"channel\":\"channel\",\"table\":\"table\",\"database\":\"db\",\"schema\":\"schema\",\"is_iceberg\":%s}";
+
+    apiOverride.addSerializedJsonOverride(
+        DROP_CHANNEL_ENDPOINT,
+        req -> {
+          try {
+            String expectedEntity;
+            if (isIcebergMode) {
+              expectedEntity = String.format(expectEntityFormat, "testPrefix_0", "true");
+            } else {
+              expectedEntity = String.format(expectEntityFormat, "null_0", "false");
+            }
+            String requestEntityString =
+                ServiceResponseHandler.consumeAndReturnResponseEntityAsString(
+                    ((HttpPost) req).getEntity());
```

Comment:
Not sure I understand what this test is doing - you're sending a request entity into a method typically used to handle a response entity?
Which means you're not really testing product flow, and really only testing what happens when EntityUtils.toString is called on the request entity.

I think your test can be equally effective with just the following:
1. In executeWithRetries method (that takes IStreamingIngestRequest), extract the code that converts the request into a string into its own public static method (`serializeRequestToString`, say). This code blob i mean:
```
    try {
      payloadInString = objectMapper.writeValueAsString(payload);
    } catch (JsonProcessingException e) {
      throw new SFException(e, ErrorCode.BUILD_REQUEST_FAILURE, message);
    }
```
2. In this testcase, directly call this new method and validate that the response does not have stringForLogging as a key.
3. Undo the change to serviceResponseHandler.